apps/service_define/lib/define/event/handler.ex

```diff
@@ -0,0 +1,18 @@
+defmodule Define.Event.Handler do
+  use Brook.Event.Handler
+  require Logger
+
+  import Events, only: [extract_start: 0, transform_define: 0, load_start: 0]
+
+  def handle_event(%Brook.Event{type: extract_start(), data: %Extract{} = extract}) do
+    Define.Event.Store.update_definition(extract)
+  end
+
+  def handle_event(%Brook.Event{type: transform_define(), data: %Transform{} = transform}) do
+    Define.Event.Store.update_definition(transform)
+  end
+
+  def handle_event(%Brook.Event{type: load_start(), data: %Load{} = load}) do
+    Define.Event.Store.update_definition(load)
+  end
```

Comment:
```suggestion
  @events [extract_start(), transform_define(), load_start()]
  
  def handle_event(%Brook.Event{type: event, data: data}) when event in @events do
    Define.Event.Store.update_definition(data)
  end
```
Just seems like a place to DRY up.

Reply:
Do you lose a secondary check that those event types have the right structs though?  Does that matter?

Reply:
Take a look at `apps/definition_events/lib/events.ex`.

This macro creates send_* functions for the events that ensure event type and event struct are locked together.